File: src/test/java/ru/uris/test/TestServer.java
package ru.uris.test;

import ru.uris.Packet;
import ru.uris.Server;

import java.io.IOException;
import java.util.Arrays;

public class TestServer {
    public static void main(String[] args) throws IOException {
        try (var server = new Server(2014)) {
            server.objectPool.add(new Object());
            server.objectPool.add(new TestClass(12));

            server.objectPool.add(TestServer.class.getClassLoader());
            var connection = server.accept();
            connection.createListener().start();

            System.out.println(Arrays.toString(((Packet.PObjectList) connection.sendAndReceive(new Packet.PObjectList(Packet.nextId()))).objects));
            System.out.println(Arrays.toString(((Packet.PMethodList) connection.sendAndReceive(new Packet.PMethodList(Packet.nextId(), 0, true))).methods));
            System.out.println();
            for (var method : ((Packet.PMethodList) connection.sendAndReceive(new Packet.PMethodList(Packet.nextId(), 0, true))).methods)
                System.out.println("[NAME]" + method.name + "\n[ARGS]:" + Arrays.toString(method.args) + "\n[RETURN]:" + method.ret + "\n");
        }
    }
}

File: src/test/kotlin/ru/kuris/test/TestServer.kt
package ru.kuris.test

import kotlinx.coroutines.runBlocking
import ru.kuris.Packet
import ru.kuris.Packet.Companion.nextId
import ru.kuris.Server

object TestServer {
    @JvmStatic
    fun main(args: Array<String>) = runBlocking {
        Server(2014).use { server ->
            server.objectPool.add(Any())
            server.objectPool.add(TestClass(12))
            server.objectPool.add(TestServer::class.java.classLoader)
            val connection = server.accept()
            connection.createListener().start()
            println((connection.sendAndReceive(Packet.Companion.PObjectList(nextId())) as Packet.Companion.PObjectList).objects.contentToString())
            println(
                (connection.sendAndReceive(
                    Packet.Companion.PMethodList(
                        nextId(),
                        0,
                        true
                    )
                ) as Packet.Companion.PMethodList).methods.contentToString()
            )
            println()
            for (method in (connection.sendAndReceive(
                Packet.Companion.PMethodList(
                    nextId(),
                    0,
                    true
                )
            ) as Packet.Companion.PMethodList).methods)
                println(
                    """
                    [NAME]${method.name}
                    [ARGS]:${method.args.contentToString()}
                    [RETURN]:${method.ret}

                    """.trimIndent()
                )
        }
    }
}